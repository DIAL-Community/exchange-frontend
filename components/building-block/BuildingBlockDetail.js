import { useRef, useState } from 'react'
import { useRouter } from 'next/router'
import { useApolloClient, useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'
import Breadcrumb from '../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../shared/GraphQueryHandler'
import { BUILDING_BLOCK_DETAIL_QUERY, BUILDING_BLOCK_POLICY_QUERY } from '../shared/query/buildingBlock'
import { fetchOperationPolicies } from '../utils/policy'
import BuildingBlockDetailLeft from './BuildingBlockDetailLeft'
import BuildingBlockDetailRight from './BuildingBlockDetailRight'

const BuildingBlockDetail = ({ slug }) => {
  const scrollRef = useRef(null)
  const client = useApolloClient()

  const router = useRouter()
  const { locale } = router

  const [editingAllowed, setEditingAllowed] = useState(false)
  const [deletingAllowed, setDeletingAllowed] = useState(false)

  const { loading, error, data } = useQuery(BUILDING_BLOCK_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.VIEWING,
        'Accept-Language': locale
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.buildingBlock) {
    return handleMissingData()
  }

  fetchOperationPolicies(
    client,
    BUILDING_BLOCK_POLICY_QUERY,
    ['editing', 'deleting']
  ).then(policies => {
    setEditingAllowed(policies['editing'])
    setDeletingAllowed(policies['deleting'])
  })

  const { buildingBlock } = data

  const slugNameMapping = () => {
    const map = {}
    map[buildingBlock.slug] = buildingBlock.name

    return map
  }

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-warm-beech text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping()}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3 shrink-0'>
          <BuildingBlockDetailLeft
            scrollRef={scrollRef}
            buildingBlock={buildingBlock}
          />
        </div>
        <div className='lg:basis-2/3'>
          <BuildingBlockDetailRight
            ref={scrollRef}
            buildingBlock={buildingBlock}
            editingAllowed={editingAllowed}
            deletingAllowed={deletingAllowed}
          />
        </div>
      </div>
    </div>
  )
}

export default BuildingBlockDetail
