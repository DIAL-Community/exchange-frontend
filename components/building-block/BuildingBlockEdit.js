import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../lib/apolloClient'
import Breadcrumb from '../shared/Breadcrumb'
import { handleLoadingQuery, handleMissingData, handleQueryError } from '../shared/GraphQueryHandler'
import { BUILDING_BLOCK_DETAIL_QUERY } from '../shared/query/buildingBlock'
import BuildingBlockEditLeft from './BuildingBlockEditLeft'
import BuildingBlockForm from './fragments/BuildingBlockForm'

const BuildingBlockEdit = ({ slug }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { loading, error, data } = useQuery(BUILDING_BLOCK_DETAIL_QUERY, {
    variables: { slug },
    context: {
      headers: {
        ...GRAPH_QUERY_CONTEXT.EDITING
      }
    }
  })

  if (loading) {
    return handleLoadingQuery()
  } else if (error) {
    return handleQueryError(error)
  } else if (!data?.buildingBlock) {
    return handleMissingData()
  }

  const { buildingBlock } = data

  const slugNameMapping = () => {
    const map = {
      edit: format('app.edit')
    }
    map[buildingBlock.slug] = buildingBlock.name

    return map
  }

  return (
    <div className='lg:px-8 xl:px-56 flex flex-col'>
      <div className='px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40'>
        <Breadcrumb slugNameMapping={slugNameMapping()}/>
      </div>
      <div className='flex flex-col lg:flex-row gap-x-8'>
        <div className='lg:basis-1/3 shrink-0'>
          <BuildingBlockEditLeft buildingBlock={buildingBlock} />
        </div>
        <div className='lg:basis-2/3'>
          <BuildingBlockForm buildingBlock={buildingBlock} />
        </div>
      </div>
    </div>
  )
}

export default BuildingBlockEdit
