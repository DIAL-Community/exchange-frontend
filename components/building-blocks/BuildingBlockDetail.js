import { useEffect, useRef } from 'react'
import { gql, useQuery } from '@apollo/client'
import NotFound from '../shared/NotFound'
import { Error, Loading } from '../shared/FetchStatus'
import BuildingBlockDetailLeft from './BuildingBlockDetailLeft'
import BuildingBlockDetailRight from './BuildingBlockDetailRight'

const BUILDING_BLOCK_QUERY = gql`
  query BuildingBlock($slug: String!) {
    buildingBlock(slug: $slug) {
      id
      name
      slug
      imageFile
      discourseId
      specUrl
      buildingBlockDescription {
        description
        locale
      }
      workflows {
        name
        slug
        imageFile
      }
      products {
        name
        slug
        imageFile
      }
    }
  }
`

const BuildingBlockDetail = ({ slug, locale }) => {
  const { loading, error, data, refetch } = useQuery(BUILDING_BLOCK_QUERY, {
    variables: { slug: slug },
    context: { headers: { 'Accept-Language': locale } },
    skip: !slug
  })

  const discourseElement = useRef()
  const scrollToDiv = (ref) => {
    ref.current.scrollIntoView({
      behavior: 'smooth'
    })
  }

  useEffect(() => {
    refetch()
  }, [locale, refetch])

  return (
    <>
      {loading && <Loading />}
      {error && error.networkError && <Error />}
      {error && !error.networkError && <NotFound />}
      {
        data && data.buildingBlock &&
          <div className='flex flex-col lg:flex-row pb-8 max-w-catalog mx-auto max-w-catalog mx-auto'>
            <div className='relative lg:sticky lg:top-66px w-full lg:w-1/3 xl:w-1/4 h-full py-4 px-4'>
              <BuildingBlockDetailLeft
                buildingBlock={data.buildingBlock}
                discourseClick={() => scrollToDiv(discourseElement)}
              />
            </div>
            <div className='w-full lg:w-2/3 xl:w-3/4'>
              <BuildingBlockDetailRight
                buildingBlock={data.buildingBlock}
                discourseRef={discourseElement}
              />
            </div>
          </div>
      }
    </>
  )
}

export default BuildingBlockDetail
