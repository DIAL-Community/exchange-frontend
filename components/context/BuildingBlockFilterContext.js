import React, { createContext, useState } from 'react'

const BuildingBlockFilterContext = createContext()
const BuildingBlockFilterDispatchContext = createContext()

const BuildingBlockFilterProvider = ({ children }) => {
  const [showMature, setShowMature] = useState(false)
  const [sdgs, setSDGs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [categoryTypes, setCategoryTypes] = useState([])

  const [search, setSearch] = useState('')
  const [sortColumn, setSortColumn] = useState('name')
  const [sortDirection, setSortDirection] = useState('asc')

  const setSdgs = setSDGs

  const buildingBlockFilterValues = {
    showMature, sdgs, useCases, workflows, categoryTypes, search, sortColumn, sortDirection
  }
  const buildingBlockFilterDispatchValues = {
    setShowMature,
    setSDGs,
    setSdgs,
    setUseCases,
    setWorkflows,
    setCategoryTypes,
    setSearch,
    setSortColumn,
    setSortDirection
  }

  return (
    <BuildingBlockFilterContext.Provider value={buildingBlockFilterValues}>
      <BuildingBlockFilterDispatchContext.Provider value={buildingBlockFilterDispatchValues}>
        {children}
      </BuildingBlockFilterDispatchContext.Provider>
    </BuildingBlockFilterContext.Provider>
  )
}

export { BuildingBlockFilterProvider, BuildingBlockFilterContext, BuildingBlockFilterDispatchContext }
