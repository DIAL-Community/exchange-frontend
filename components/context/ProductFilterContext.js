import { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [isEndorsed, setIsEndorsed] = useState(false)
  const [productDeployable, setProductDeployable] = useState(false)
  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [tags, setTags] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])
  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])

  const [search, setSearch] = useState('')

  const [pageNumber, setPageNumber] = useState(0)
  const [pageOffset, setPageOffset] = useState(0)

  const productFilterValues = {
    isEndorsed,
    productDeployable,
    origins,
    countries,
    sectors,
    organizations,
    sdgs,
    tags,
    useCases,
    workflows,
    buildingBlocks,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showGovStackOnly,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    search,
    pageOffset,
    pageNumber
  }

  const productFilterDispatchValues = {
    setIsEndorsed,
    setProductDeployable,
    setOrigins,
    setCountries,
    setSectors,
    setOrganizations,
    setSdgs,
    setTags,
    setUseCases,
    setWorkflows,
    setBuildingBlocks,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowGovStackOnly,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setSearch,
    setPageNumber,
    setPageOffset
  }

  return (
    <ProductFilterContext.Provider value={productFilterValues}>
      <ProductFilterDispatchContext.Provider value={productFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
