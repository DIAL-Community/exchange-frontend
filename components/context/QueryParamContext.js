import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
