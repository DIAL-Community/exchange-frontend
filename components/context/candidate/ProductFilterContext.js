import React, { createContext, useState } from 'react'

const ProductFilterContext = createContext()
const ProductFilterDispatchContext = createContext()

const ProductFilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')

  const candidateProductFilterValues = {
    search
  }
  const candidateProductFilterDispatchValues = {
    setSearch
  }

  return (
    <ProductFilterContext.Provider value={candidateProductFilterValues}>
      <ProductFilterDispatchContext.Provider value={candidateProductFilterDispatchValues}>
        {children}
      </ProductFilterDispatchContext.Provider>
    </ProductFilterContext.Provider>
  )
}

export { ProductFilterProvider, ProductFilterContext, ProductFilterDispatchContext }
