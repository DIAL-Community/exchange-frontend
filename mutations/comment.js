import { gql } from '@apollo/client'

export const CREATE_COMMENT = gql`
  mutation CreateComment(
    $commentId: String!
    $commentObjectType: String!
    $commentObjectId: Int!
    $text: String!
    $userId: Int!
    $parentCommentId: String
  ) {
    createComment(
      commentId: $commentId
      commentObjectType: $commentObjectType
      commentObjectId: $commentObjectId
      text: $text
      userId: $userId
      parentCommentId: $parentCommentId
    ) {
      errors
    }
  }
`
export const DELETE_COMMENT = gql`
  mutation DeleteComment($commentId: String!) {
    deleteComment(commentId: $commentId) {
      errors
    }
  }
`
