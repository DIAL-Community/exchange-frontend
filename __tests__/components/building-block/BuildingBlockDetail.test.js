import { screen } from '@testing-library/dom'
import BuildingBlockDetail from '../../../components/building-block/BuildingBlockDetail'
import BuildingBlockEdit from '../../../components/building-block/BuildingBlockEdit'
import { BuildingBlockFilterProvider } from '../../../components/context/BuildingBlockFilterContext'
import { ProductFilterProvider } from '../../../components/context/ProductFilterContext'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { BUILDING_BLOCK_DETAIL_QUERY } from '../../../components/shared/query/buildingBlock'
import { COMMENTS_QUERY } from '../../../components/shared/query/comment'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import { buildingBlockComments, buildingBlockDetail } from './data/BuildingBlockDetail.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the building block main page.', () => {
  const mockBuildingBlockPaginationAttribute = generateMockApolloData(
    BUILDING_BLOCK_DETAIL_QUERY,
    {
      'slug': 'analytics-and-business-intelligence'
    },
    null,
    buildingBlockDetail
  )

  const mockPaginatedBuildingBlocks = generateMockApolloData(
    COMMENTS_QUERY,
    {
      'commentObjectId': 25,
      'commentObjectType': 'BUILDING_BLOCK'
    },
    null,
    buildingBlockComments
  )

  test('Should render detail of a building block.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockPaginatedBuildingBlocks, mockBuildingBlockPaginationAttribute]}>
        <QueryParamContextProvider>
          <BuildingBlockFilterProvider>
            <ProductFilterProvider>
              <BuildingBlockDetail slug='analytics-and-business-intelligence' />
            </ProductFilterProvider>
          </BuildingBlockFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Analytics and business intelligence')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  test('Should render unathorized for non logged in user.', async () => {
    const { container } = render(
      <CustomMockedProvider mocks={[mockPaginatedBuildingBlocks, mockBuildingBlockPaginationAttribute]}>
        <QueryParamContextProvider>
          <BuildingBlockFilterProvider>
            <ProductFilterProvider>
              <BuildingBlockEdit slug='analytics-and-business-intelligence' />
            </ProductFilterProvider>
          </BuildingBlockFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Analytics and business intelligence')).toBeInTheDocument()
    expect(await screen.findByText('You are not authorized to view this page')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
