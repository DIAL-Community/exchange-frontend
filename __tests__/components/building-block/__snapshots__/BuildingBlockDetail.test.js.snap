// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the building block main page. Should render detail of a building block. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-warm-beech text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-ochre font-semibold"
              >
                Analytics and business intelligence
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="w-20 h-20 px-1 py-1 rounded-full bg-dial-orange"
                >
                  <img
                    alt="Logo of Building Block."
                    class="object-contain dial-ochre-filter w-14 h-14 mx-auto my-2 white-filter"
                    src="http://localhost:3000//assets/building-blocks/analytics-and-business-intelligence.svg"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex gap-x-3"
                >
                  <div
                    class="flex flex-col gap-y-3 text-sm grow shrink-0"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Maturity
                    </div>
                    <div
                      class="flex text-dial-stratos"
                    >
                      DRAFT
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-y-3 text-sm grow shrink-0"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Category
                    </div>
                    <div
                      class="flex text-dial-stratos"
                    >
                      FUNCTIONAL
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="flex flex-col gap-y-3 text-sm py-3"
            >
              <div
                class="font-semibold text-dial-ochre"
              >
                Jump To
              </div>
              <div
                class=" css-1uaxf62-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-:r0:-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="react-select__control css-afrqkl-control"
                >
                  <div
                    class="react-select__value-container css-17t7gkf-ValueContainer"
                  >
                    <div
                      class="react-select__placeholder css-5rjclb-placeholder"
                      id="react-select-:r0:-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="react-select__input-container css-qbdosj-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-describedby="react-select-:r0:-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-label="Current active navigation"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class="react-select__input"
                        id="async-select-id-undefined"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-nyxz9h-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-2hgo35-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
            <div
              class="hidden lg:flex flex-col gap-y-3"
            >
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-lg text-dial-sapphire font-semibold"
                >
                  Bookmark
                </div>
                <button
                  class="group flex flex-row gap-x-3"
                  type="button"
                >
                  <img
                    alt="Logo of Bookmark."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/bookmark-icon.svg"
                    width="40"
                  />
                  <div
                    class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                  >
                    <div
                      class="text-sm text-dial-stratos text-left"
                    >
                      Bookmark this page.
                    </div>
                  </div>
                </button>
                <div
                  class="text-sm text-dial-stratos"
                >
                  Your bookmarks can be found in your Account page.
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-lg text-dial-sapphire font-semibold"
                >
                  Share
                </div>
                <div
                  class="flex flex-row gap-3"
                >
                  <img
                    alt="Logo of Twitter."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/twitter-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of LinkedIn."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/linkedin-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of Email."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/email-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of Copy."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/copy-icon.svg"
                    width="40"
                  />
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-lg text-dial-sapphire font-semibold"
                >
                  Comment
                </div>
                <div
                  class="text-sm text-dial-stratos"
                >
                  
    Have a question for the community on this Building Block or have some insight you would like to share?
  
                </div>
                <div
                  class="flex text-white"
                >
                  <div
                    class="bg-dial-iris-blue rounded-md text-sm"
                  >
                    <button
                      class="px-5 py-3"
                      type="button"
                    >
                      Post a comment
                    </button>
                  </div>
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="px-4 lg:px-0 py-4 lg:py-6"
        >
          <div
            class="flex flex-col gap-y-3"
          >
            <div
              class="text-xl font-semibold text-dial-ochre py-3"
            >
              Description
            </div>
            <div
              class="block"
            >
              <div
                class="relative"
              >
                <div
                  class="html-react-parser-viewer text-base"
                >
                  Building block description goes here.
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-blue-chalk my-3"
            />
            <div
              class="flex flex-col gap-y-3"
            >
              <div
                class="flex flex-col gap-y-3"
              >
                <div
                  class="flex flex-row gap-3"
                >
                  <div
                    class="text-xl font-semibold text-dial-ochre"
                  >
                    Products
                  </div>
                  <div
                    class="flex gap-3 ml-auto"
                  />
                </div>
                <div
                  class="text-xs text-justify italic text-dial-stratos mb-2"
                >
                  Products that fulfil the technical specifications of this building block.
                </div>
                <div
                  class="grid grid-cols-1 lg:grid-cols-2 gap-x-8 gap-y-3"
                >
                  <div>
                    <div
                      class="relative"
                    >
                      <a
                        href="/products/agrinfo-jembe"
                      >
                        <div
                          class="rounded-lg bg-gradient-to-r from-product-bg-light to-product-bg h-16"
                        >
                          <div
                            class="flex flex-row gap-x-3 px-6 h-full"
                          >
                            <img
                              alt="Logo of Products."
                              class="object-contain w-10 h-10 my-auto min-w-[2.5rem]"
                              src="http://localhost:3000//assets/products/agrinfo-jembe.png"
                            />
                            <div
                              class="text-sm font-semibold my-auto text-dial-meadow line-clamp-2"
                            >
                              AgrInfo Jembe
                            </div>
                          </div>
                        </div>
                      </a>
                      <div
                        class="absolute top-2 right-2"
                      />
                    </div>
                  </div>
                  <div>
                    <div
                      class="relative"
                    >
                      <a
                        href="/products/goldkeys"
                      >
                        <div
                          class="rounded-lg bg-gradient-to-r from-product-bg-light to-product-bg h-16"
                        >
                          <div
                            class="flex flex-row gap-x-3 px-6 h-full"
                          >
                            <img
                              alt="Logo of Products."
                              class="object-contain w-10 h-10 my-auto min-w-[2.5rem]"
                              src="http://localhost:3000//assets/products/goldkeys.png"
                            />
                            <div
                              class="text-sm font-semibold my-auto text-dial-meadow line-clamp-2"
                            >
                              Goldkeys
                            </div>
                          </div>
                        </div>
                      </a>
                      <div
                        class="absolute top-2 right-2"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-blue-chalk my-3"
            />
            <div
              class="flex flex-col gap-y-3"
            >
              <div
                class="flex flex-col gap-y-3"
              >
                <div
                  class="flex flex-row gap-3"
                >
                  <div
                    class="text-xl font-semibold text-dial-ochre"
                  >
                    Workflows
                  </div>
                  <div
                    class="flex gap-3 ml-auto"
                  />
                </div>
                <div
                  class="grid grid-cols-1 lg:grid-cols-2 gap-x-8 gap-y-3"
                >
                  <div>
                    <div
                      class="relative"
                    >
                      <a
                        href="/workflows/data-analysis-and-business-intelligence"
                      >
                        <div
                          class="rounded-lg bg-gradient-to-r from-workflow-bg-light to-workflow-bg h-16"
                        >
                          <div
                            class="flex flex-row gap-x-3 px-6 h-full"
                          >
                            <div
                              class="rounded-full bg-dial-plum w-10 h-10 my-auto min-w-[2.5rem]"
                            >
                              <img
                                alt="Logo of Workflows."
                                class="object-contain w-6 h-6 mx-auto mt-2 white-filter"
                                src="http://localhost:3000//assets/workflows/data-analysis-and-business-intelligence.svg"
                              />
                            </div>
                            <div
                              class="text-sm font-semibold text-dial-plum my-auto"
                            >
                              Data Analysis and Business Intelligence
                            </div>
                          </div>
                        </div>
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-blue-chalk my-3"
            />
            <div
              class="lg:hidden flex flex-col gap-y-3"
            >
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-lg text-dial-sapphire font-semibold"
                >
                  Bookmark
                </div>
                <button
                  class="group flex flex-row gap-x-3"
                  type="button"
                >
                  <img
                    alt="Logo of Bookmark."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/bookmark-icon.svg"
                    width="40"
                  />
                  <div
                    class="my-auto border-b border-transparent group-hover:border-dial-slate-500"
                  >
                    <div
                      class="text-sm text-dial-stratos text-left"
                    >
                      Bookmark this page.
                    </div>
                  </div>
                </button>
                <div
                  class="text-sm text-dial-stratos"
                >
                  Your bookmarks can be found in your Account page.
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
              <div
                class="flex flex-col gap-3 py-3"
              >
                <div
                  class="text-lg text-dial-sapphire font-semibold"
                >
                  Share
                </div>
                <div
                  class="flex flex-row gap-3"
                >
                  <img
                    alt="Logo of Twitter."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/twitter-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of LinkedIn."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/linkedin-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of Email."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/email-icon.svg"
                    width="40"
                  />
                  <img
                    alt="Logo of Copy."
                    class="object-contain"
                    height="40"
                    src="/ui/v1/copy-icon.svg"
                    width="40"
                  />
                </div>
              </div>
              <hr
                class="border-b border-dial-slate-200"
              />
            </div>
            <div
              class="text-dial-sapphire"
            >
              <div
                id="comments-section"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Unit tests for the building block main page. Should render unathorized for non logged in user. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="lg:px-8 xl:px-56 flex flex-col"
  >
    <div
      class="px-4 lg:px-6 py-4 bg-dial-blue-chalk text-dial-stratos ribbon-detail z-40"
    >
      <div
        class="whitespace-nowrap text-ellipsis overflow-hidden"
      >
        <a
          class="h5"
          href="/"
        >
          Home
        </a>
      </div>
    </div>
    <div
      class="flex flex-col lg:flex-row gap-x-8"
    >
      <div
        class="lg:basis-1/3"
      >
        <div
          class="bg-dial-slate-100 lg:h-full"
        >
          <div
            class="flex flex-col gap-y-3 px-4 lg:px-6 lg:py-3"
          >
            <div
              class="flex flex-col gap-y-4 py-3"
            >
              <div
                class="text-xl text-dial-ochre font-semibold"
              >
                Analytics and business intelligence
              </div>
              <div
                class="flex justify-center items-center py-16 bg-white rounded border"
              >
                <div
                  class="w-20 h-20 px-1 py-1 rounded-full bg-dial-orange"
                >
                  <img
                    alt="Logo of Building Block."
                    class="object-contain dial-ochre-filter w-14 h-14 mx-auto my-2 white-filter"
                    src="http://localhost:3000//assets/building-blocks/analytics-and-business-intelligence.svg"
                  />
                </div>
              </div>
              <div
                class="flex flex-col gap-y-8 text-sm pt-6 pb-3"
              >
                <div
                  class="flex gap-x-3"
                >
                  <div
                    class="flex flex-col gap-y-3 text-sm grow shrink-0"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Maturity
                    </div>
                    <div
                      class="flex text-dial-stratos"
                    >
                      DRAFT
                    </div>
                  </div>
                  <div
                    class="flex flex-col gap-y-3 text-sm grow shrink-0"
                  >
                    <div
                      class="font-semibold text-dial-sapphire"
                    >
                      Category
                    </div>
                    <div
                      class="flex text-dial-stratos"
                    >
                      FUNCTIONAL
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="border-b border-dial-slate-200"
            />
          </div>
        </div>
      </div>
      <div
        class="lg:basis-2/3"
      >
        <div
          class="h-[60vh]"
        >
          <div
            class="flex bg-dial-alice-blue h-full"
          >
            <div
              class="text-dial-stratos text-lg w-full my-auto flex flex-col gap-4"
            >
              <svg
                class="mx-auto"
                fill="currentColor"
                height="3em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="3em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zm0-384c13.3 0 24 10.7 24 24V264c0 13.3-10.7 24-24 24s-24-10.7-24-24V152c0-13.3 10.7-24 24-24zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
                />
              </svg>
              <div
                class="text-center mt-5"
              >
                You are not authorized to view this page
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
